src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py

```diff
@@ -556,19 +502,17 @@ def _generate_python_command(
             "--num_hosts=$SLURM_NTASKS",
             "--server_addr=$SLURM_JOB_MASTER_NODE:12345",
             "--host_idx=$SLURM_PROCID",
-            f"--job_log_dir={cmd_args[f'{self.test_name}.setup_flags.docker_workspace_dir']}",
-            f"--tfds_data_dir={cmd_args[f'{self.test_name}.setup_flags.tfds_data_dir']}",
-            f"--enable_checkpoint_saving={cmd_args[f'{self.test_name}.setup_flags.enable_checkpoint_saving']}",
+            f"--job_log_dir={cmd_args[f'{self.test_name}.setup_flags']['docker_workspace_dir']}",
+            f"--tfds_data_dir={cmd_args[f'{self.test_name}.setup_flags']['tfds_data_dir']}",
+            f"--enable_checkpoint_saving={cmd_args[f'{self.test_name}.setup_flags']['enable_checkpoint_saving']}",
             "--multiprocess_gpu",
             "--alsologtostderr",
             f'--fdl_config="{fdl_config}"',
         ]
 
-        # Dynamically adding fdl. prefixed arguments
-        fdl_prefix = f"{self.test_name}.fdl."
-        fdl_args = {k[len(fdl_prefix) :]: v for k, v in cmd_args.items() if k.startswith(fdl_prefix)}
+        fdl_args: dict[str, str] = cmd_args[f"{self.test_name}.fdl"]
 
-        for key, value in fdl_args.items():
+        for key, value in sorted(fdl_args.items()):
```

Comment:
There is no specific requirement for ordering of fiddle flags or it should be parsed in a specific order? Any reason why this change is needed here?

Reply:
Main reason to add that was to simplify generated scripts comparison against current state. For that I also made similar changes in `main` during my tests.

As a side affect, it should be easier to track CLI options inside our scripts if that is required.

To finalise, the reason is to simplify development.

Reply:
Can we have the generated test script in the test plan? There is lot of changes in this PR esp with the fdl flags and xla flags. These flags basically are needed for performance convergence. I trust you when you say it’s easier to parse. But want to see the generated scripts. Maybe your cloudaix PR should have it?

Reply:
That is what I did. I run mentioned in description commands to generate run scripts. Did it for these changes and for main state. Then compared file to file. Same for Cloud AI X, just commands were different. If you think it is needed, I will of course upload these files.

You can also run this version for your configs and see if everything is in place. As I mentioned, I didn't run all the scenarios, but for every found issue I added a unit test.